pkg/webui/console/views/application/index.js

```diff
@@ -79,6 +80,7 @@ import PropTypes from '../../../lib/prop-types'
     loadData: id => {
       dispatch(getApplication(id, 'name,description'))
       dispatch(getApplicationsRightsList(id))
+      dispatch(getApplicationDeviceCount(id))
```

Comment:
This call is only relevant if the user goes to the overview page. In case the user has no rights to view the devices the request will result in a redundant request with `403` code. We can detect this beforehand.

You can move this call to the application overview page, something like this:
```diff
@ pkg/webui/console/views/application-overview/connect.js:28 @ import {
  selectCollaboratorsTotalCount,
  selectCollaboratorsFetching,
} from '../../store/selectors/collaborators'
+ import { getApplicationDeviceCount } from '../../store/actions/applications'
import { selectApiKeysTotalCount, selectApiKeysFetching } from '../../store/selectors/api-keys'
import { getCollaboratorsList } from '../../store/actions/collaborators'
import { getApiKeysList } from '../../store/actions/api-keys'
@ pkg/webui/console/views/application-overview/connect.js:85 @ const mapDispatchToProps = dispatch => ({
    mayViewApplicationCollaborators,
    mayViewApplicationApiKeys,
    mayViewApplicationLink,
+   mayViewDevices,
    appId,
  ) {
    if (mayViewApplicationCollaborators) dispatch(getCollaboratorsList('application', appId))
    if (mayViewApplicationApiKeys) dispatch(getApiKeysList('application', appId))
    if (mayViewApplicationLink) dispatch(getApplicationLink(appId))
+    if (mayViewDevices) dispatch(getApplicationDeviceCount(appId))
  },
})

@ pkg/webui/console/views/application-overview/connect.js:104 @ const mergeProps = (stateProps, dispatchProps, ownProps) => ({
      stateProps.mayViewApplicationCollaborators,
      stateProps.mayViewApplicationApiKeys,
      stateProps.mayViewApplicationLink,
+     stateProps.mayViewDevices,
      stateProps.appId,
    ),
})
```